Feature/create sign up form

Ahmed-Fahmy-Dev opened the pull request.

Ahmed-Fahmy-Dev committed:
install needed packages:
- mui date pickers
- date library

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into feature/create-signUp-form

Ahmed-Fahmy-Dev committed:
create index files
- create index for this feature components
- create index for this feature data

Ahmed-Fahmy-Dev committed:
create form data
- validation schema
- initial values
- form model

Ahmed-Fahmy-Dev committed:
create the form itself:
- parent sign up form wrapper
- user info form
- login info form
- succesfull sign up modal
- adding signup form to the signup page

Ahmed-Fahmy-Dev committed:
fixed Swiper.scss:
- file formatting fixed which caused error in StyleLint rules

Ahmed-Fahmy-Dev committed:
fixing formatting  in App scss file

Ahmed-Fahmy-Dev committed:
Swiper.scss fix

Ahmed-Fahmy-Dev committed:
test commit

AltysS commented on client/src/features/signup/components/SignupForm/SignupForm.js:
```diff
@@ -0,0 +1,88 @@
+      setSubmitting(false);
+    }, 2000);
+
+    // console.log(values);
```
This console log can be removed

Ahmed-Fahmy-Dev replied:
its for future reference, in order to know what kind of object we are sending to the API and adjust it accordingly

AltysS commented on client/src/features/signup/components/LoginInfoForm/LoginInfoForm.js:
```diff
@@ -0,0 +1,36 @@
+        </Grid>
+
+        <Grid item xs={12} laptop={6}>
+          <Field component={TextField} name={login.name} label={login.label} fullWidth required />
```
Here there is one mistake. Once I clicked on any field and do not type any data, I receive an error at each field even if I don't touch it. As far as I remember, we can write a custom input and receive props *touched* inside this component. Please, review the screen below. Here I receive props inside the Formik component and have access to the touched field. Guess it`s better to fix this issue here

![image](https://user-images.githubusercontent.com/112622259/202896668-71201d8c-94bc-4863-a7ac-6da3cdb1d580.png)
![image](https://user-images.githubusercontent.com/112622259/202896695-7c726252-e4dc-4823-8986-c5d38c6db871.png)

Ahmed-Fahmy-Dev replied:
am aware of it,  this functionality will be fixed in future updates as this branch is just for JSX  and styling

AltysS commented on client/src/features/signup/components/LoginInfoForm/LoginInfoForm.js:
```diff
@@ -0,0 +1,36 @@
+        </Grid>
+
+        <Grid item xs={12} laptop={6}>
+          <Field component={TextField} name={password.name} label={password.label} fullWidth required />
```
I think that it`s better to hide the password here.
![image](https://user-images.githubusercontent.com/112622259/202896787-1cc80165-a390-446c-b5eb-53bf2bd4af0c.png)

Ahmed-Fahmy-Dev replied:
agree, this will be performed in a future update including the mui InputAdornment component

AltysS commented on client/src/features/signup/components/UserInfoForm/UserInfoForm.js:
```diff
@@ -0,0 +1,46 @@
+        User Info
+      </Typography>
+      <Grid container spacing={3} sx={{ mb: 3 }}>
+        <Grid item xs={12} laptop={6}>
```
I guess that phone, gender and and birthday has to be required field as you have it at your props component
![image](https://user-images.githubusercontent.com/112622259/202897369-5b1fa2ed-5b16-42fa-a90b-2352a648148b.png)

Ahmed-Fahmy-Dev committed:
snapshot tests written:
- Login form
- Signup form
- Signup success
- UserInfo form
- Signup Page

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into feature/create-signUp-form

Ahmed-Fahmy-Dev committed:
hotfix: snapshot tests
- added mocked component to signupForm test
- removed birthdate initial value and updated test

Ahmed-Fahmy-Dev committed:
Merge branch 'develop' into feature/create-signUp-form

Ahmed-Fahmy-Dev committed:
hotfix: fix SliderButton import path

HIMDanny commented on client/src/features/signup/components/LoginInfoForm/__snapshots__/LoginInfoForm.test.js.snap:
```diff
@@ -0,0 +1,191 @@
+  <div>
+    <div>
+      <div
+        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
```
I suppose you've missed mocking components

Ahmed-Fahmy-Dev replied:
nope.  that is the formik-mui input field.  jest understands it and no conflicts occur in testing if left unmocked.

HIMDanny commented on client/src/features/signup/components/UserInfoForm/UserInfoForm.js:
```diff
@@ -0,0 +1,46 @@
+        </Grid>
+
+        <Grid item xs={12} laptop={4}>
+          <Field component={DatePicker} disableFuture name={birthDate.name} label={birthDate.label} fullWidth />
```
I noticed it works the same with or without `fullWidth` prop, so rather delete it

Ahmed-Fahmy-Dev replied:
true.. done.

HIMDanny commented on client/src/features/signup/data/formModel.js:
```diff
@@ -0,0 +1,59 @@
+
+  retypePassword: {
+    name: 'retypePassword',
+    label: 'Retype Password',
```
if I'm not mistaken it should be `Repeat Password`

Ahmed-Fahmy-Dev replied:
nope. its Re-enter or Re-type actually )

HIMDanny commented on client/src/features/signup/components/SignupForm/SignupForm.js:
```diff
@@ -0,0 +1,88 @@
+                          position: 'absolute',
+                          top: '50%',
+                          left: '50%',
+                          marginTop: '-15px',
+                          marginLeft: '-15px',
```
I think it is better to use translate to properly center it
```suggestion
                          translate: '-50% -50%',
```

Ahmed-Fahmy-Dev replied:
actually i dont know why but i tried using translate and transform translate as am used to but its not working in this case,  when i checked the component in mui documents, i found they are using the margin left and top properties.

HIMDanny replied:
I've tried it before suggesting to you) Everything works fine with translate

HIMDanny reviewed (commented):
Suggest deleting bottom margin from radio button label
![image](https://user-images.githubusercontent.com/58943798/203130025-4cd0c4fa-0334-4887-84a7-f690b219fa71.png)

How about changing day with month?
![image](https://user-images.githubusercontent.com/58943798/203130464-dfd16b82-c0ee-4c7e-a33f-26bba23ad2d5.png)

How about hiding "Confirm" on loading?
![image](https://user-images.githubusercontent.com/58943798/203131448-cc2b59c7-e716-41c2-8e93-1f4f6e4ed6fe.png)

This error happens, when I try to type something in fields
![image](https://user-images.githubusercontent.com/58943798/203133884-f7ea19d2-72a3-4e86-a8c6-f907a0ca300d.png)

This error happens when success modal is shown
![image](https://user-images.githubusercontent.com/58943798/203134204-d95fe17c-f3f8-41ed-b34d-132044759d2a.png)

Ahmed-Fahmy-Dev committed:
fixed console error
- added string value to success prop

Ahmed-Fahmy-Dev committed:
spelling correction in label

Ahmed-Fahmy-Dev committed:
fix : nested component style fix
- radio button span margin bottom removed

Ahmed-Fahmy-Dev committed:
fixed and updates to form:
- changed date format to dd/mm/yyy by adding enAu date locale
- removed loading state as it is not needed
- submit and back button dissappear when form is submitted and loading

Ahmed-Fahmy-Dev committed:
fix:  removing un needed styles:
- removed position absolute styling from CircularProgress

Ahmed-Fahmy-Dev committed:
fix: FormControlLabel styling:
- removed sx prop
- utilized componentsProps to remove bottom margin from label span
